.github/workflows/pull-request.yaml

```diff
@@ -59,3 +59,18 @@ jobs:
       - uses: actions/checkout@v2
       - run: npm install @apidevtools/swagger-cli
       - run: node_modules/.bin/swagger-cli validate orderbook/openapi.yml
+  coverage:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install Rust
+        run: rustup toolchain install nightly --component llvm-tools-preview
+      - name: Install cargo-llvm-cov
+        run: curl -LsSf https://github.com/taiki-e/cargo-llvm-cov/releases/latest/download/cargo-llvm-cov-x86_64-unknown-linux-gnu.tar.gz | tar xzf - -C ~/.cargo/bin
+      - name: Generate code coverage
+        run: cargo llvm-cov --all-features --lcov --output-path lcov.info
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v2
+        with:
+          files: lcov.info
+          fail_ci_if_error: true
```

Comment:
Maybe a bit harsh to fail on upload failure, but can readjust if necessary.

Reply:
AFAIU, the `codecov` CI step is not required for merging - but if it fails to upload it will show an `X` on the status so we are aware. So the result I expected with this (although I may have misunderstood) would be that it would show the `codecov` step as failed, but the PR would still be allowed to merge as long as the other required steps (`openapi` and `rust-*` steps) are green.